### ISSUE TITLE
Clean up static PuffinApp references in window classes

### ISSUE THREAD

**Issue by pont-us** [pont-us/PuffinPlot#204]
Many of the classes in the window subpackage use  PuffinApp.getInstance() to find the PuffinPlot application. It would probably be cleaner to have the PuffinApp instance supplied to their constructors.

**Comment by pont-us** [comment]
```
Bugs Everywhere data:
Created at: 2011-12-20 19:12:45
Status: open
Severity: minor
UUID: 55ca6c0c-e2b8-4bc8-bfc6-c1d10639efb3
Short name: a88/55c
Ditz bug index: 202
```

**Comment by pont-us** [comment]
I'm not sure exactly when this was fixed, but certainly at the latest by commit 1a17ca5a606db, in which the `PuffinApp.getInstance()` method itself was removed.
